org.eclipse.xtend.core/src/org/eclipse/xtend/core/jvmmodel/XtendJvmModelInferrer.java

```diff
@@ -391,10 +392,14 @@ protected void initialize(XtendClass source, JvmGenericType inferredJvmType) {
 			if (typeRefToObject != null)
 				inferredJvmType.getSuperTypes().add(typeRefToObject);
 		} else {
-			inferredJvmType.getSuperTypes().add(jvmTypesBuilder.cloneWithProxies(extendsClause));
+			JvmTypeReference superClass = jvmTypesBuilder.cloneWithProxies(extendsClause);
```

Comment:
API wise:
What do you think about 
```
jvmTypesBuilder.addImplementedInterface(inferredJvmType, superInterface);
jvmTypesBuilder.setSuperClass(inferredJvmType, superClass);
```
Internally that would do `cloneWithProxies` and add the adapter.

Reply:
I like it! :)
I would call the first one `addSuperInterface` though since for an interface `implemented` is misleading: in that case it should be `extended`. With `SuperIterface` we could cover both cases, right?